scripts/webpack.client.config.js

```diff
@@ -7,102 +7,83 @@ const HtmlWebpackPlugin = require("html-webpack-plugin");
-        rel: "preload",
-        as: "script"
-      }),
-      new WebpackPwaManifest({
-        name: "Coder",
-        short_name: "Coder",
-        description: "Run VS Code on a remote server",
-        background_color: "#e5e5e5",
-        icons: [
-          {
-            src: path.join(root, "packages/web/assets/logo.png"),
-            sizes: [96, 128, 192, 256, 384]
-          }
-        ]
-      })
-    ].concat(prod ? [
-      new GenerateSW({
-        exclude: [/\.map$/, /^manifest.*\.js$/, /\.html$/],
-        runtimeCaching: [
-          {
-            urlPattern: new RegExp("^(?!.*(html))"),
-            handler: "StaleWhileRevalidate",
-            options: {
-              cacheName: "code-server",
-              expiration: {
-                maxAgeSeconds: 86400
-              },
-              cacheableResponse: {
-                statuses: [0, 200]
-              }
-            }
-          }
-          // Network first caching is also possible.
-          /*{
-        urlPattern: new RegExp("^(?!.*(html))"),
-        handler: "NetworkFirst",
-        options: {
-          networkTimeoutSeconds: 4,
-          cacheName: "code-server",
-          expiration: {
-            maxAgeSeconds: 86400,
-          },
-          cacheableResponse: {
-            statuses: [0, 200],
-          },
-        },
-      }*/
-        ]
-      })
-    ] : [new webpack.HotModuleReplacementPlugin()]),
-    target: "web"
-  });
+	require("./webpack.general.config")(options), {
+	devtool: prod ? "none" : "cheap-module-eval-source-map",
+	mode: prod ? "production" : "development",
+	entry: prod ? options.entry : [
+		"webpack-hot-middleware/client?reload=true&quiet=true",
+		options.entry,
+	],
+	module: {
+		rules: [{
+			test: /\.s?css$/,
+			// This is required otherwise it'll fail to resolve CSS in common.
+			include: root,
+			use: [{
+				loader: MiniCssExtractPlugin.loader,
+			}, {
+				loader: "css-loader",
+			}, {
+				loader: "sass-loader",
+			}],
+		}, {
+			test: /\.(png|ttf|woff|eot|woff2)$/,
+			use: [{
+				loader: "file-loader",
+				options: {
+					name: "[path][name].[ext]",
+				},
+			}],
+		}, {
+			test: /\.svg$/,
+			loader: 'url-loader'
+		}],
+	},
+	plugins: [
+		new MiniCssExtractPlugin({
+			chunkFilename: `${options.name || "client"}.[name].[hash:6].css`,
+			filename: `${options.name || "client"}.[name].[hash:6].css`
+		}),
+		new HtmlWebpackPlugin({
+			template: options.template
+		}),
+		new PreloadWebpackPlugin({
+			rel: "preload",
+			as: "script"
+		}),
+		new WebpackPwaManifest({
+			name: "Coder",
+			short_name: "Coder",
+			description: "Run VS Code on a remote server",
+			background_color: "#e5e5e5",
+			icons: [{
+				src: path.join(root, "packages/web/assets/logo.png"),
+				sizes: [96, 128, 192, 256, 384],
+			}],
+		})
+	].concat(prod ? [
+		// Exclude everything from the precache and add everything we want to be
+		// cached at runtime.
+		new GenerateSW({
+			exclude: [/.*/],
```

Comment:
We don't precache the bundle anymore now... before it could cache this and other assets while still on the login screen. This made the loading after the login significantly faster. Why exactly do you exclude all now? Maybe explicitly set it to cache the same files as in the runtimeCaching pattern.

Reply:
Yeah, I was deliberating on this but I think you're right and it's probably better to load them all up at once at the beginning.